lms/djangoapps/shoppingcart/tests/test_models.py

```diff
@@ -375,6 +375,19 @@ def test_generate_receipt_instructions_callchain(self):
             cart.generate_receipt_instructions()
             mock_gen_inst.assert_called_with()
 
+    def test_confirmation_email_error(self):
+        cart = Order.get_cart_for_user(self.user)
+        CertificateItem.add_to_order(cart, self.course_key, self.cost, 'honor')
+
+        # Simulate an error when sending the confirmation
+        # email.  This should NOT raise an exception.
+        # If it does, then the implicit view-level
+        # transaction could cause a roll-back, effectively
+        # reversing order fulfillment.
+        with patch.object(mail.message.EmailMessage, 'send') as mock_send:
+            mock_send.side_effect = Exception("Kaboom!")
+            cart.purchase()
+
```

Comment:
Do you want to assert that the purchase actually was committed as well by reading the Order back and checking state?